mobx/decorators.js

```diff
@@ -18,10 +18,10 @@ import {action, observable} from 'mobx';
 export function settable(target, property, descriptor) {
     const name = 'set' + upperFirst(property);
     if (!target.hasOwnProperty(name)) {
-        const fn = action.bound(target, name, {
-            value: function(v) {this[property] = v}
-        });
-        Object.defineProperty(target, name, fn);
+        const value = function(v) {
+            runInAction(() => {this[property] = v});
+        };
+        Object.defineProperty(target, name, {value});
```

Comment:
The new code here seems more clear to me - curious if this was required by mobx changes or just to learn more about when we discuss.

Reply:
was required due to a change in how mobx decorators work, specifically `action`